impala/thrift_sasl.py

```diff
@@ -20,20 +17,46 @@
 #
 """ SASL transports for Thrift. """
 
-# Copied from the Impala repo
+# Initially copied from the Impala repo
 
-import sys
+from __future__ import absolute_import
 
-from cStringIO import StringIO
-from thrift.transport import TTransport
-from thrift.transport.TTransport import *
-from thrift.protocol import TBinaryProtocol
-try:
-  import saslwrapper as sasl
-except ImportError:
-  import sasl
+import sys
 import struct
 
+import sasl
+
+
+PY2 = sys.version_info[0] == 2
+PY3 = sys.version_info[0] == 3
+
+
+if PY2:
+  from cStringIO import StringIO
+  from thrift.transport.TTransport import (
```

Comment:
Yeah, this is the only exceptional case, bc this file has been copied around from multiple repos.  I'll likely contribute these changes back to the main impala repo and Hue.